src/v/config/property.h

```diff
@@ -119,4 +119,43 @@ class one_or_many_property : public property<std::vector<T>> {
     }
 };
 
+/**
+ * A numeric property that is clamped to a range.
+ */
+template<typename T>
+class clamped_property : public property<T> {
+public:
+    using property<T>::property;
+
+    clamped_property(
+      config_store& conf,
+      std::string_view name,
+      std::string_view desc,
+      required req = required::yes,
+      T def = T{},
+      std::optional<T> min = std::nullopt,
+      std::optional<T> max = std::nullopt)
+      : property<T>(conf, name, desc, req, def)
+      , _min(min)
+      , _max(max) {}
+
+    void set_value(YAML::Node n) override {
+        auto val = std::move(n.as<T>());
+
+        if (val.has_value()) {
+            if (_min.has_value()) {
+                val = std::max(val, _min.value());
+            }
+            if (_max.has_value()) {
+                val = std::min(val, _max.value());
+            }
+        }
+        this->_value = std::move(val);
+    };
+
+private:
+    std::optional<T> _min;
```

Comment:
Seems like could be templetized and known at compile time since it is declared at config.cc 

not sure worth spending the time on, but seems like no need for runtime storage.